### ISSUE TITLE
Internal error "unevaluated expression" compiling array of element accesses at O2

### ISSUE THREAD

**Issue by sbronson** [MiniZinc/libminizinc#536]
```
array [1..1] of var 0..0: a;
constraint [a[1]][1]=0;
```
```
$ minizinc -c -O2 f.mzn
MiniZinc has encountered an internal error. This is a bug.
Please file a bug report using the MiniZinc bug tracker.
The internal error message was:
"unevaluated expression"
```

On develop.

**Comment by cyderize** [comment]
This appears to have been fixed in 2.6.0.

**Comment by cyderize** [comment]
Ah sorry, I didn't realise this was with -O2. This bug is still present.
